frontend/src/pages/AllArchievedAgreements.tsx

```diff
@@ -0,0 +1,70 @@
+import AddIcon from '@mui/icons-material/Add';
+import img from '../assets/Group_120.png';
+import { useTranslation } from 'react-i18next';
+import { StringBank } from '../strings';
+import { useNavigate } from 'react-router-dom';
+import { useContext } from 'react';
+import { Button, Stack, Typography } from '@mui/material';
+import { AgreementCarousel } from 'components';
+import { GroupContext } from 'contexts/group';
+import EmptyArchivePage from './EmptyArchivePage';
+
+const AllArchievedAgreements = () => {
+  const navigate = useNavigate();
+  const { t } = useTranslation();
+  const handleMenuItemClick = (e: React.MouseEvent<HTMLElement>, slug = '') => {
+    navigate(`/${slug}/new-agreement`);
+  };
+  const { archivedAgreements, name, slug } = useContext(GroupContext);
+  if (archivedAgreements?.length > 0) {
+    return (
+      <Stack
+        justifyContent={archivedAgreements?.length ? 'start' : 'space-between'}
+        height="100%"
+        padding={{ sm: 2 }}
+        spacing={2}
+      >
+        <Stack flexDirection="row" justifyContent="space-between" paddingX={1}>
+          <Typography variant="h2">{t(StringBank.GROUP_AGREEMENTS, { group: name })}</Typography>
+          <Button
+            variant="contained"
+            startIcon={<AddIcon />}
+            onClick={(event) => handleMenuItemClick(event, slug)}
+          >
+            {t(StringBank.NEW_AGREEMENT)}
+          </Button>
+        </Stack>
+        {archivedAgreements.length ? (
```

Comment:
We don't need to check length again. We only need to return the agreements gallery (carousel), because we're inside length > 0

Reply:
removed